codec/src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -0,0 +1,40 @@
+package io.netty.handler.codec.compression;
+
+import com.nixxcode.jvmbrotli.enc.BrotliOutputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+
+public class BrotliEncoder extends MessageToByteEncoder<ByteBuf> {
+
+    private final int totalLength;
+    private int writtenLength;
+
+    public BrotliEncoder(int totalLength) {
+        this.totalLength = totalLength;
+    }
+
+    private BrotliOutputStream compressorOS;
+    private ByteBufOutputStream byteBufOutputStream;
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, ByteBuf msg, ByteBuf out) throws Exception {
+        if (compressorOS == null) {
+            byteBufOutputStream = new ByteBufOutputStream(out);
+            compressorOS = new BrotliOutputStream(byteBufOutputStream);
+        }
+
+        writtenLength += msg.readableBytes();
+        compressorOS.write(ByteBufUtil.getBytes(msg));
+
+        if (writtenLength == totalLength) {
+            byteBufOutputStream.close();
+            compressorOS.close();
+        } else {
+            byteBufOutputStream.flush();
```

Comment:
What's the reasoning behind flushing per message?

Reply:
Support chunked data.

Reply:
Well assuming say 4kB or 64kB uncompressed messages, if they compress nicely, do you want to flush on the boundary of each incoming uncompressed message? Would it be more optimal to let it automatically flush based on the new compressed message size?

Reply:
Manual Flush is needed else we won't get data in `ByteBuf`.